packages/web-app/src/hooks/useCachedDao.tsx

```diff
@@ -0,0 +1,92 @@
+import {isAddress} from 'ethers/lib/utils';
+import {useCache} from './useCache';
+import {useClient} from './useClient';
+import {
+  ICreateDaoERC20Voting,
+  ICreateDaoWhitelistVoting,
+} from '@aragon/sdk-client';
+import {Dao} from 'utils/types';
+
+interface IUseDaoResponse {
+  update: (dao: Dao) => Dao;
+  createErc20: (dao: ICreateDaoERC20Voting) => Promise<string>;
+  createWhitelist: (dao: ICreateDaoWhitelistVoting) => Promise<string>;
+  get: (address: string) => Dao;
+}
+
+export const useDao = (): IUseDaoResponse => {
+  const {get: getCache, set: setCache} = useCache();
+  const {erc20: erc20Client, whitelist: whitelistClient} = useClient();
+  const get = (address: string): Dao => {
+    if (!isAddress(address)) {
+      throw Error('invalid address when trying to get dao');
+    }
+    const cacheKey = `dao-${address}`;
+    const cacheValue = getCache(cacheKey);
+    // If dao is in cache return it
+    if (cacheValue) {
+      return cacheValue;
+    }
+    // client.getDao(address).then((value) => setCache(cacheKey, value)).catch((e)=> throw Error(e))
```

Comment:
Will it be useful in the future?

Reply:
I'm not sure, this is part of a `get by address` function that I left there as a placeholder when first implemented the`useCachedDao` hook. There is also an `update` placeholder, I can get rid of both of them if you want to clean the hook

Reply:
I'd remove it, if this is so trivial to work out when needed.

BTW, you cannot `throw Error(...)`, you should either `throw new Error("text here")` or else `thow e`. But then, you don't need the `catch` block:
```suggestion
    // client.getDao(address).then((value) => setCache(cacheKey, value)).catch((e)=> throw e)
```